aten/src/ATen/ParallelNative.cpp

```diff
@@ -275,7 +275,7 @@ bool in_parallel_region() {
 #endif // C10_MOBILE
 }
 
-void intraop_launch(std::function<void()> func) {
+void intraop_launch(const std::function<void()>& func) {
 #ifndef C10_MOBILE
   if (!in_parallel_region() && get_num_threads() > 1) {
     _get_intraop_pool().run(std::move(func));
```

Comment:
These should be by value considering it gets forwarded by C10_MOBILE code path. Be careful, clang-tidy won't check all paths of IFDEF

Reply:
After some thought, I think we should not change those thread pool API and I have reverted the related commits